crates/sui-core/src/state_accumulator.rs

```diff
@@ -118,33 +119,44 @@ impl StateAccumulator {
             .flat_map(|fx| {
                 fx.unwrapped()
                     .iter()
-                    .map(|(oref, _owner)| (oref.0, oref.1))
-                    .collect::<Vec<(ObjectID, SequenceNumber)>>()
+                    .map(|(oref, _owner)| (*fx.transaction_digest(), oref.0, oref.1))
+                    .collect::<Vec<(TransactionDigest, ObjectID, SequenceNumber)>>()
             })
             .chain(effects.iter().flat_map(|fx| {
                 fx.unwrapped_then_deleted()
                     .iter()
-                    .map(|oref| (oref.0, oref.1))
-                    .collect::<Vec<(ObjectID, SequenceNumber)>>()
+                    .map(|oref| (*fx.transaction_digest(), oref.0, oref.1))
+                    .collect::<Vec<(TransactionDigest, ObjectID, SequenceNumber)>>()
             }))
-            .collect::<Vec<(ObjectID, SequenceNumber)>>();
+            .collect::<Vec<(TransactionDigest, ObjectID, SequenceNumber)>>();
 
-        let unwrapped_ids: HashSet<ObjectID> =
-            all_unwrapped.iter().map(|(id, _)| id).cloned().collect();
+        let unwrapped_ids: HashMap<TransactionDigest, Vec<ObjectID>> = all_unwrapped
+            .iter()
+            .map(|(digest, id, _)| (*digest, *id))
+            .collect::<Vec<(TransactionDigest, ObjectID)>>()
+            .into_iter()
```

Comment:
pretty sure collect() followed by into_iter() is unnecessary

Reply:
Yeah thanks, meant to clean this up